lib/node_modules/@stdlib/blas/base/dscal/lib/dscal.js

```diff
@@ -43,39 +43,10 @@ var M = 5;
 * // x => <Float64Array>[ -10.0, 5.0, 15.0, -25.0, 20.0, 0.0, -5.0, -15.0 ]
 */
 function dscal( N, alpha, x, stride ) {
-	var i;
-	var m;
-
 	if ( N <= 0 || stride <= 0 || alpha === 1.0 ) {
 		return x;
 	}
-	// Use loop unrolling if the stride is equal to `1`...
-	if ( stride === 1 ) {
-		m = N % M;
-
-		// If we have a remainder, run a clean-up loop...
-		if ( m > 0 ) {
-			for ( i = 0; i < m; i++ ) {
-				x[ i ] *= alpha;
-			}
-		}
-		if ( N < M ) {
-			return x;
-		}
-		for ( i = m; i < N; i += M ) {
-			x[ i ] *= alpha;
-			x[ i+1 ] *= alpha;
-			x[ i+2 ] *= alpha;
-			x[ i+3 ] *= alpha;
-			x[ i+4 ] *= alpha;
-		}
-		return x;
-	}
-	N *= stride;
-	for ( i = 0; i < N; i += stride ) {
-		x[ i ] *= alpha;
-	}
-	return x;
+	return ndarray( N, alpha, x, stride, 0 );
```

Comment:
@aman-095 I believe we want to support negative strides here. That means using `stride2offset` to compute the offset before calling into the `ndarray` API.

Reply:
But, native `dscal` implementation from `lapack` doesn't allow for negative stride.

Reply:
That was my point about converting from negative to positive. But here, our `ndarray.js` file should support negative strides. In C, we should convert the stride before calling CBLAS. Meaning, *we* are free to support negative strides, but we need to ensure that if and when we call a library following LAPACK that we only pass along positive strides. In short, for all our Level 1 routines, we should be consistent in supporting negative strides. The current situation of some supporting and others not supporting is just confusing.

Reply:
It means that for our Js implementation, we should allow negative strides for all BLAS Level 1 and only for `CBLAS`, or other hardware-optimized libraries we use the same convention of stride conversion.

Reply:
Correct. We need to update our JS implementations to support negative strides.